CMakeLists.txt

```diff
@@ -228,6 +228,10 @@ if (${TRITON_PYTORCH_DOCKER_BUILD})
     COMMAND /bin/sh -c "if [ -f libmkl_avx2.so ]; then patchelf --add-needed libmkl_core.so libmkl_avx2.so; fi"
     COMMAND /bin/sh -c "if [ -f libmkl_avx512.so ]; then patchelf --add-needed libmkl_gnu_thread.so libmkl_avx512.so; fi"
     COMMAND /bin/sh -c "if [ -f libmkl_avx512.so ]; then patchelf --add-needed libmkl_core.so libmkl_avx512.so; fi"
+    COMMAND /bin/sh -c "if [ -f libmkl_vml_def.so ]; then patchelf --add-needed libmkl_gnu_thread.so libmkl_vml_def.so; fi"
```

Comment:
Add a comment describing why we are patchelf'ing these shared libraries here for the future reference.

Reply:
And how to obtain the dependency graph.

Reply:
Done

Reply:
@tanmayv25 obtaining the dependency graph is complicated and a matter of reviewing the mkl library and its inner workings. 

I have skipped adding the logic here since it requires using ldd, LD_DEBUG, nm and understanding how mkl works.